android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -450,14 +466,16 @@ constructor(
             title = subject,
             indicatorTitle = subject,
             measureReportDenominator =
-              if (indicators.size == 1) indicators.first().count.toInt() else null,
-            dataList = if (indicators.size > 1) indicators else emptyList()
+              if (theIndicators.size == 1) theIndicators.first().count.toInt() else null,
+            dataList = if (theIndicators.size > 1) theIndicators else emptyList()
           )
         )
       }
 
     measureReports
-      .takeIf { it.all { it.type != MeasureReport.MeasureReportType.INDIVIDUAL } }
+      .takeIf {
+        it.all { measureReport -> measureReport.type != MeasureReport.MeasureReportType.INDIVIDUAL }
+      }
       ?.forEach { report ->
```

Comment:
I think we can avoid a loop by pushing the takeIf into the forEach as a conditional, if this is something kotlin will optimize on compile time, then nevermind, but if not this should save time

```suggestion
      .forEach { report ->
        if (report.type == MeasureReport.MeasureReportType.INDIVIDUAL) return@forEach
```